lib/account/account-service.js

```diff
@@ -49,7 +40,9 @@ module.exports = function (config, secureHash, Storage) {
         return getDirectoryAsync(registrationDirName, regId).then(getAsync).then((accountFile) => {
             accountFile.password = accountInfo.password;
 
-            return makeAccountFileAsync(accountFile, options).then(storage.delAsync);
+            return makeAccountFileAsync(accountInfo.accountId, accountFile, options).then(() => {
+                storage.delAsync;
```

Comment:
You said the tests pass.  This means you should first fix the tests to detect this error and then fix the code.